src/alfalfa/AlfalfaComponent.hpp

```diff
@@ -12,8 +12,7 @@ namespace alfalfa {
   class ALFALFA_API AlfalfaComponent
   {
    public:
-    template <typename T, std::enable_if_t<std::is_base_of<AlfalfaComponentBase, T>::value, bool> = true>
-    AlfalfaComponent(T component) : m_component(std::make_unique<T>(std::move(component))) {}
+    AlfalfaComponent(const AlfalfaComponentBase& component) : m_component(component.clone()) {}
```

Comment:
Curious why you added this one in the main source but the others just in the Alfalfa.i

Reply:
I didn't want the ctor exposed to the bindings so i added it here and ignored the ctors for the class in the .i .